packages/core/src/odata-common/request-builder/batch/batch-response-parser.spec.ts

```diff
@@ -258,4 +261,63 @@ describe('batch response parser', () => {
       });
     });
   });
+
+  describe('parseBatchResponse', () => {
+    it('parses a batch response with both empty and non-empty bodies', () => {
+      const data = [
+        '--3B17E95920A7FAF8BCB7495D043515000',
+        'Content-Type: multipart/mixed; boundary=3B17E95920A7FAF8BCB7495D043515001',
+        'Content-Length:      2427',
+        '',
+        '--3B17E95920A7FAF8BCB7495D043515001',
+        'Content-Type: application/http',
+        'Content-Length: 71,',
+        'content-transfer-encoding: binary',
+        '',
+        'HTTP/1.1 204 No Content',
+        'Content-Length: 0',
+        'dataserviceversion: 2.0',
+        '',
+        '{"valid": "json"}',
+        '',
+        '',
+        '--3B17E95920A7FAF8BCB7495D043515001',
```

Comment:
```suggestion
        '--3B17E95920A7FAF8BCB7495D043515001',
```
I tried to build a batch request like below by using our cap e2e application, it seems there is no empty lines between the response body and the next boundary.
```
batch(
      getAll,
      changeset(create,deleteRequestBuilder)
    )
```